hadoop-crypto/src/test/java/com/palantir/crypto2/hadoop/StandaloneEncryptedFileSystemTest.java

```diff
@@ -100,46 +100,60 @@ public void testReadWrite() throws IOException {
         assertThat(readData).isNotEqualTo(dataBytes);
 
         // KeyMaterial file exists
-        assertThat(rawFs.exists(new Path(path + FileKeyStorageStrategy.EXTENSION))).isTrue();
+        assertThat(rawFs.exists(keyMaterialPath(path))).isTrue();
     }
 
     @Test
     public void testDelete() throws IOException {
         File rootFolder = folder.newFolder();
 
         Path path1 = writeData(rootFolder);
-        Path path2 = writeData(rootFolder);
 
         assertThat(efs.exists(path1)).isTrue();
-        assertThat(efs.exists(path2)).isTrue();
+        assertThat(rawFs.exists(path1)).isTrue();
+        assertThat(rawFs.exists(keyMaterialPath(path1))).isTrue();
 
         efs.delete(new Path(path1.toString()), false);
 
         assertThat(efs.exists(path1)).isFalse();
-        assertThat(efs.exists(path2)).isTrue();
+        assertThat(rawFs.exists(path1)).isFalse();
+        assertThat(rawFs.exists(keyMaterialPath(path1))).isFalse();
     }
 
     @Test
     public void testRecursiveDelete() throws IOException {
         File rootFolder = folder.newFolder();
+        Path rootPath = new Path(rootFolder.getAbsolutePath());
 
         Path path1 = writeData(rootFolder);
         Path path2 = writeData(rootFolder);
 
         assertThat(efs.exists(path1)).isTrue();
+        assertThat(rawFs.exists(path1)).isTrue();
+        assertThat(rawFs.exists(keyMaterialPath(path1))).isTrue();
+
         assertThat(efs.exists(path2)).isTrue();
+        assertThat(rawFs.exists(path2)).isTrue();
+        assertThat(rawFs.exists(keyMaterialPath(path2))).isTrue();
+
+        efs.delete(rootPath, true);
 
-        efs.delete(new Path(rootFolder.getAbsolutePath()), true);
+        assertThat(rawFs.exists(rootPath)).isFalse();
 
         assertThat(efs.exists(path1)).isFalse();
+        assertThat(rawFs.exists(path1)).isFalse();
+        assertThat(rawFs.exists(keyMaterialPath(path1))).isFalse();
+
         assertThat(efs.exists(path2)).isFalse();
+        assertThat(rawFs.exists(path2)).isFalse();
+        assertThat(rawFs.exists(keyMaterialPath(path2))).isFalse();
     }
 
     private Path writeData(File rootFolder) throws IOException {
         // Write encrypted data
         java.nio.file.Path filePath = Files.createTempFile(rootFolder.toPath(), "prefix", "suffix");
         Path newPath = new Path(filePath.toAbsolutePath().toString());
-        OutputStream os = efs.create(path);
```

Comment:
oh, this is majestic

Reply:
was wondering why the non-recursive delete worked before ...

Reply:
well the file actually does get created by `Files.createTempFile` above which made this even more insidious